src/AWS.Deploy.Orchestration/DisplayedResources/DisplayedResourcesFactory.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using AWS.Deploy.Orchestration.Data;
+
+namespace AWS.Deploy.Orchestration.DisplayedResources
+{
+    /// <summary>
+    /// Interface for displayed resources such as <see cref="ElasticBeanstalkEnvironmentResource"/>
+    /// </summary>
+    public interface IDisplayedResourceCommand
+    {
+        Task<Dictionary<string, string>> Execute(string resourceId);
+    }
+
+    public interface IDisplayedResourcesFactory
+    {
+        IDisplayedResourceCommand? GetResource(string resourceType);
+    }
+
+    /// <summary>
+    /// Factory class responsible to build and get the displayed resources.
+    /// </summary>
+    public class DisplayedResourcesFactory : IDisplayedResourcesFactory
```

Comment:
`IDisplayedResourceCommandFactory` and `DisplayedResourceCommandFactory`

Reply:
Done